### PR TITLE
dev: comment and cleanup environment.yml

### PR DESCRIPTION
**Pull request by jandom**
Lots of discussion in https://github.com/MDAnalysis/UserGuide/pull/293 big thanks to @IAlibay and @lilyminium for helping me out. 

This PR removes a few packages and adds a lot of comments. It's still possible that there are other unused packages.

The aim of this PR is to make it clearer which dep is needed where, it's still not super granular, and it's still possible that other packages are not needed.

### CODE REVIEW

File: environment.yml
Comment (lilyminium) [inline environment.yml]: I mostly think of these as sphinx "extensions" -- we use the `ipython` directive in some docs, `nbsphinx` to render notebooks, etc. If they're imported/used in `conf.py` it would generally be in their role as extensions.

File: environment.yml
Comment (lilyminium) [inline environment.yml]: I would label this section generally `extensions` instead. `nbconvert` especially I would assume is not called by any notebook explicitly but to download notebooks, or in nbsphinx.

File: environment.yml
Comment (lilyminium) [inline environment.yml]: As we edit or update notebooks, I can see the notebook-specific labels getting outdated quite easily. Instead, could we maybe organise these by purpose? e.g. `plotly`, `ipygany`, `pyvista`, `scikit-image`, `seaborn` are for visualising/plotting; `moviepy` and `nglview` are for visualising/making movies of molecules; `dask` and `distributed` you could switch the heading to `# parallelization`.

**Review by lilyminium — CHANGES_REQUESTED** [review]
Thanks @jandom for reorganising this file! I had some comments on the conceptual grouping. I'm not sure organising by notebook file is easiest to understand and maintain, especially since there's nothing stopping us from using e.g. `scikit-image` in another notebook and forgetting to update this. Instead, I've suggested ways to document by overall purpose.

File: environment.yml
Comment (IAlibay) [inline environment.yml]: It's odd that we only cover a small subset of the MDA setup and install requirements:

setup requirements: [Cython, packaging, numpy>=1.22.3, setuptools, wheel]
additional install requirements: [biopython>=1.8, networkx>=2.0, GridDataFormats>=0.4.0, mmtf-python>=1.0.0, joblib>=0.12, matplotlib>=1.5.1, tqdm>=4.43.0, threadpoolctl, fasterners]

File: environment.yml
Comment (IAlibay) [inline environment.yml]: this is available on conda-forge now: https://github.com/conda-forge/pyedr-feedstock

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: environment.yml
Comment (lilyminium) [reply environment.yml]: These might be covered by the `pip` install of `develop` MDA -- iirc only cython and numpy are crucial to have before installing MDA. At least CI is correctly building from this environment. I'd lean towards having as few dependencies here as possible so if core MDA updates its dependencies, it's reflected here as well.

Comment (IAlibay) [reply environment.yml]: Even cython and numpy aren't crucial. They get picked up on wheel creation (the magic of our pyproject.toml). I'd advocate for getting rid of all the non optional dependencies then (although it does add to the risk of a dependency resolution failure).

File: environment.yml
Comment (jandom) [reply environment.yml]: Thank you these changes were easy to apply

File: environment.yml
Comment (jandom) [reply environment.yml]: Thanks, yeah, easy to change

File: environment.yml
Comment (jandom) [reply environment.yml]: Changed the grouping here as well, easy to apply

File: environment.yml
Comment (jandom) [reply environment.yml]: i'm assuming you wanted this changed to a conda package? or was it just a "fyi"?

File: environment.yml
Comment (jandom) [reply environment.yml]: I'm not really sure what's the request here? I've removed numpy and cython from the environment.yml, i'm happy to proceed and remove other deps assuming the CI/CD passes – what's needed here?

File: environment.yml
Comment (lilyminium) [inline environment.yml]: Could you move this down to #visualisation?

File: environment.yml
Comment (lilyminium) [reply environment.yml]: Thanks @jandom -- could you please move these up to the #extensions then?

File: environment.yml
Comment (lilyminium) [inline environment.yml]:
```suggestion
      # docs themes
```

File: environment.yml
Comment (lilyminium) [inline environment.yml:51]:
```suggestion
      # extensions
      - sphinx-sitemap
```

**Review by lilyminium — CHANGES_REQUESTED** [review]
Hi @jandom, thank you for making the changes! Just a couple things would be nice to re-arrange and then I think we're good to go.

File: environment.yml
Comment (lilyminium) [reply environment.yml]: Personally I'd be happy with as minimal MDA requirements as possible here.

Comment (jandom) [reply environment.yml]: Agreed but it'll take a lot of iteration to remove everything but what's absolutely needed

File: environment.yml
Comment (lilyminium) [reply environment.yml]: Just this to go -- nbsphinx, ipywidgets and the like are used to render notebooks, and IIRC ipython to use the `ipython` sphinx directive. If they're imported in conf.py it's to configure settings but, otherwise they fall into the same category as other extensions, e.g. sphinx bibtex.

**Review by lilyminium — CHANGES_REQUESTED** [review]

**Comment by lilyminium** [comment]
I see @IAlibay is sick -- did you still want to re-visit this PR, Irfan? If not I can take over if you dismiss your review or approve changes.

**Comment by IAlibay** [comment]
As far as I am aware (edit: from looking at my last request changes) there are still four build or core dependencies of
MDAnalysis left in the yaml file (matplotlib, networkx, packaging and
threadpoolctl). If those are included intentionally then it would be great
to document why, if not it would be good to remove them here.

If this PR needs a quick resolution then please go ahead and dismiss my
review.

**Comment by jandom** [comment]
@lilyminium this should be resolved now, re extensions. I still have it in me to try and remove all the other packages that are MDA deps

File: environment.yml
Comment (IAlibay) [inline environment.yml]:
```suggestion
```

File: environment.yml
Comment (IAlibay) [inline environment.yml]:
```suggestion
```

File: environment.yml
Comment (IAlibay) [inline environment.yml]:
```suggestion
```

File: environment.yml
Comment (IAlibay) [inline environment.yml]:
```suggestion
```

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: environment.yml
Comment (jandom) [reply environment.yml]: Let's try these all, thanks!

**Comment by jandom** [comment]
Sorry a history question... please ignore if it's a time waste... but i'm really curious how all these MDA deps made their way into here?

**Review by IAlibay — APPROVED** [review]

**Comment by jandom** [comment]
@IAlibay thanks for the confidence and just nuking those deps – this helped tremendously, I would have been removing them one-by-one, because I don't have the context

**Comment by jandom** [comment]
@lilyminium PTAL, this seems to be blocked on an approval from you – and i don't think it merits an override (it's nor urgent)

**Comment by IAlibay** [comment]
> Sorry a history question... please ignore if it's a time waste... but i'm really curious how all these MDA deps made their way into here?

They've been around since before we started supporting PEP518, i.e. a world before you could just rely on pip to pick up the right build time dependencies & before we offered wheels. Back then it was safer to have your own deps before you attempted to pip install something.

**Review by lilyminium — APPROVED** [review]
Sorry for the wait -- thanks for the changes @jandom and for all your patience!

**Comment by jandom** [comment]
Thank you for all you reviews 👏

**Comment by jandom** [comment]
Looking back at this month, we got a lot done – my original aim was to wrap up the snapshot tests and [refactor PR](https://github.com/MDAnalysis/UserGuide/pull/301), and I think we can do that too. @IAlibay and @lilyminium this wouldn't be possible without your engagement and encouragement, thank you!

**Pull request merged**
